### PR TITLE
Update storage checker log to provide clearer stats

### PR DESCRIPTION
**Pull request by jackjlli**
This PR updates storage checker log to provide clearer statistics.

Previously the estimated size is just existing size + newly incoming compress segment size, and it's hard for user to understand.

This changes explicitly show the static data(storage quota, num replicas), dynamic data (existing quota usage), and newly incoming data(new segment compressed size).

### CODE REVIEW

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/validation/StorageQuotaChecker.java [hunk 1]
```diff
@@ -111,21 +111,28 @@ public QuotaCheckerResponse isSegmentStorageWithinQuota(@Nonnull File segmentFil
...
+    long existingSegmentSizeBytes = sizeDetails != null ? sizeDetails.estimatedSizeInBytes : 0;
 
     // Since tableNameWithType comes with the table type(OFFLINE), thus we guarantee that
     // tableSubtypeSize.estimatedSizeInBytes is the offline table size.
-    _controllerMetrics.setValueOfTableGauge(tableName, ControllerGauge.OFFLINE_TABLE_ESTIMATED_SIZE, tableSubtypeSize.estimatedSizeInBytes);
+    _controllerMetrics.setValueOfTableGauge(tableName, ControllerGauge.OFFLINE_TABLE_ESTIMATED_SIZE,
+        tableSubtypeSize.estimatedSizeInBytes);
 
-    long estimatedFinalSizeBytes = tableSubtypeSize.estimatedSizeInBytes - existingSegmentSizeBytes + incomingSegmentSizeBytes;
+    // incomingSegmentSizeBytes is compressed data size for just 1 replica. Dictionary and indices sizes aren't included.
```
Comment (sunithabeeram) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/validation/StorageQuotaChecker.java]: Dictionary/indices may be included. Remove the last part of this comment.

File: pinot-controller/src/test/java/com/linkedin/pinot/controller/validation/StorageQuotaCheckerTest.java [hunk 2]
```diff
@@ -106,7 +105,10 @@ public void testWithinQuota() throws IOException, InvalidConfigException {
     try (FileOutputStream ostr = new FileOutputStream(tempFile)) {
       ostr.write(data);
     }
-    setupTableSegmentSize(5800, 900);
+    setupTableSegmentSize(4800, 900);
+//    doReturn(_quotaConfig).when(_tableConfig.getQuotaConfig());
```
Comment (sunithabeeram) [inline pinot-controller/src/test/java/com/linkedin/pinot/controller/validation/StorageQuotaCheckerTest.java]: Remove commented out code.

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/validation/StorageQuotaChecker.java [hunk 3]
```diff
@@ -118,21 +118,23 @@ public QuotaCheckerResponse isSegmentStorageWithinQuota(@Nonnull File segmentFil
...
+          incomingSegmentSizeBytes, numReplicas, allowedStorageBytes, quotaConfig.storageSizeBytes(), numReplicas,
+          tableName);
       LOGGER.info(message);
       return new QuotaCheckerResponse(true, message);
     } else {
       String message = String.format(
-          "Estimated size: %d bytes exceeds the total configured quota of %d bytes for table %s.\nStorage quota: %s (%d bytes). Number of replicas: %d.\nExisting total usage among all replicas: %d bytes.\nIncoming compressed segment size: %d bytes.",
-          estimatedFinalSizeBytes, allowedStorageBytes, tableName, quotaConfig.getStorage(),
-          quotaConfig.storageSizeBytes(), numReplicas, tableSubtypeSize.estimatedSizeInBytes, incomingSegmentSizeBytes);
+          "Estimated size: %d bytes ( = current usage among all replicas: %d bytes - (existing segment sizes among all replicas: %d bytes) + (incoming compressed segment size: %d bytes * number of replicas: %d)) exceeds the total configured quota of %d bytes ( = configured quota: %d bytes * number of replicas: %d ) for table %s.",
```
Comment (sunithabeeram) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/validation/StorageQuotaChecker.java]: Lets first tell the user about the problem and then elaborate. "Storage quota exceeded. Estimated size....".

On thinking more, some of the error cases happen where the table has already exceeded size (indexes enabled prior to push etc). In that case, I think we can probably say, "Table already over quota. Estimated uncompressed size of all replicas > %d (configured value X * num replicas). Check if indexes were enabled recently and adjust table quota accordingly".

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/validation/StorageQuotaChecker.java [hunk 4]
```diff
@@ -111,21 +111,38 @@ public QuotaCheckerResponse isSegmentStorageWithinQuota(@Nonnull File segmentFil
...
 
-    long estimatedFinalSizeBytes = tableSubtypeSize.estimatedSizeInBytes - existingSegmentSizeBytes + incomingSegmentSizeBytes;
+    // incomingSegmentSizeBytes is compressed data size for just 1 replica.
+    long estimatedFinalSizeBytes =
+        tableSubtypeSize.estimatedSizeInBytes - existingSegmentSizeBytes + (incomingSegmentSizeBytes * numReplicas);
     if (estimatedFinalSizeBytes <= allowedStorageBytes) {
-      String message = String.format("Estimated size: %d bytes is within the configured quota of %d (bytes) for table %s. Incoming segment size: %d (bytes)",
-          estimatedFinalSizeBytes, allowedStorageBytes, tableName, incomingSegmentSizeBytes);
+      String message = String.format(
+          "Newly estimated size: %d bytes ( = existing estimated uncompressed size of all replicas: %d bytes - (existing segment sizes of all replicas: %d bytes) + (incoming compressed segment size: %d bytes * number replicas: %d)) is within total allowed storage size: %d bytes ( = configured quota: %d bytes * number replicas: %d) for table %s.",
```
[the remark this answers is not in the document]
Comment (mcvsubbu) [reply pinot-controller/src/main/java/com/linkedin/pinot/controller/validation/StorageQuotaChecker.java]: Maybe we should output different messages depending on append or refresh? In any case, please incluide clearly wheher the segment is in quota or not, and also the table name and segment name

**Comment by jackjlli** [comment]
https://github.com/linkedin/pinot/pull/2958 per @mcvsubbu's request

**Pull request merged**